symengine/pow.h

```diff
@@ -34,12 +34,12 @@ class Pow : public Basic
     //! \return `true` if canonical
     bool is_canonical(const Basic &base, const Basic &exp) const;
     //! \return `base` of `base**exp`
-    inline RCP<const Basic> get_base() const
+    inline const RCP<const Basic>& get_base() const
     {
         return base_;
     }
     //! \return `exp` of `base**exp`
-    inline RCP<const Basic> get_exp() const
+    inline const RCP<const Basic>& get_exp() const
```

Comment:
This is actually a problem. If it is constant reference, then it should return `const Basic &` itself, not RCP. You only return RCP if you want to assign somewhere else (and increment a refcount), which you do at several places.